### PR TITLE
Support affinity and toleration in connection pooler deployments

### LINKED ISSUE #1288: Set affinity and tolerations to logical backup cronjob
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** registry.opensource.zalan.do/acid/postgres-operator:v1.6.0
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** Bare Metal K8s
- **Are you running Postgres Operator in production?** yes
- **Type of issue?** feature request

I didn't find a way how to set affinity and tolerations into logical backup cronjob (I didn't count patch the cronjob after it is created). In my opinion, this could be really helpful if I am able to schedule all backup jobs to special node/s for his purpose.

### LINKED ISSUE #1441: Cluster affinity does not affect pooler pods
- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.6.1
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** AWS K8s
- **Are you running Postgres Operator in production?** yes
- **Type of issue?** question

Hi! I have set up cluster with nodeAffinity so that it runs on separate nodes. StatefulSet for postgres pods contains nodeAffinity section and respects it, but Deployment for poolers (...-cluster-pooler, ...-cluster-pooler-repl) does not contain this section and it's pods are deployed on a random node. Is there any config option to force poolers to run with the same nodeAffinity?

### PR DESCRIPTION
**Pull request by cucxabong**
Fix #1441 
Closes #1288

### CODE REVIEW

File: pkg/cluster/connection_pooler.go [hunk 1]
```diff
@@ -285,6 +285,26 @@ func (c *Cluster) generateConnectionPoolerPodTemplate(role PostgresRole) (
 		},
 	}
 
+	var affinity *v1.Affinity
+
+	nodeAffinity := nodeAffinity(c.OpConfig.NodeReadinessLabel, spec.NodeAffinity)
+	if c.OpConfig.EnablePodAntiAffinity {
```
Comment (FxKu) [inline pkg/cluster/connection_pooler.go:291]: can you not use `generatePodAffinity` here? Like it's done in `generatePodTemplate`? You just have to pass different labels. Or am I missing something?

Comment (cucxabong) [reply pkg/cluster/connection_pooler.go]: Thank you for your suggestion. I have updated

**Comment by frankconrad** [comment]
This is missing the part for tolerations for node affinity as well. To allow to run it on dedicated node-group/pool.

**Comment by frankconrad** [comment]
I added tolerations support here:
[postgres-operator.patch.txt](https://github.com/zalando/postgres-operator/files/6496434/postgres-operator.patch.txt)

**Comment by cucxabong** [comment]
Thank you @frankconrad , I incorporated your changes into this PR

**Comment by FxKu** [comment]
:+1:

**Comment by RafiaSabih** [comment]
:+1:

**Comment by FxKu** [comment]
Thanks @cucxabong and @frankconrad for your joint effort on this nice little enhancement.

**Comment by haslersn** [comment]
Now, on our cluster with 3 pooler instances and 3 K8s nodes, the `RollingUpdate` of the pooler Deployment is stuck, because it attempts to create a new pooler instance before terminating any.

**Pull request merged**